apps/web/pages/readonly-checkout.vue

```diff
@@ -55,85 +58,168 @@
+);
 
-  if (shippingAddresses.value.length === 0) {
-    billingAddresses.value.length > 0
-      ? await saveShippingAddress(billingAddresses.value[0])
-      : navigateTo(localePath(paths.cart));
+const fetchShippingAndPaymentMethods = async () => {
+  try {
+    await Promise.all([
+      useAggregatedCountries().fetchAggregatedCountries(),
+      getShippingMethods(),
+      fetchPaymentMethods().then(
+        async () =>
+          await savePaymentMethod(paymentMethodData?.value?.list?.find((method) => method.name === 'PayPal')?.id || 0),
+      ),
+    ]);
+  } catch (error) {
+    useHandleError(error as ApiError);
   }
+};
 
-  await useCheckoutAddress(AddressType.Shipping).set(shippingAddresses.value[0], true);
-  await useCheckoutAddress(AddressType.Billing).set(billingAddresses.value[0], true);
-  await getShippingMethods();
+const handleAuthUserInit = async () => {
+  try {
+    await useFetchAddress(AddressType.Shipping).fetchServer();
+    await persistShippingAddress();
+    await useFetchAddress(AddressType.Billing).fetchServer();
+    await persistBillingAddress();
+    await fetchShippingAndPaymentMethods();
+    setInitialCartTotal();
+  } catch (error) {
+    useHandleError(error as ApiError);
+  }
 };
 
-await getSession();
-if (cartIsEmpty.value) await navigateTo(localePath(paths.cart));
-await loadAddresses();
-await getShippingMethods();
-await fetchPaymentMethods();
-await savePaymentMethod(
-  paymentMethodData?.value?.list?.find((method: PaymentMethod) => method.name === 'PayPal')?.id ?? 0,
-);
+const setClientCheckoutAddress = async () => {
+  try {
+    await useCheckoutAddress(AddressType.Shipping).set(shippingAddresses.value[0], true);
+    await useCheckoutAddress(AddressType.Billing).set(billingAddresses.value[0], true);
+    setInitialCartTotal();
+  } catch (error) {
+    useHandleError(error as ApiError);
+  }
+};
+
+const handleGuestUserInit = async () => {
+  try {
+    await getShippingAddresses();
+    await getBillingAddresses();
+
+    if (billingAddresses.value.length === 0) await navigateTo(localePath(paths.cart));
+    if (shippingAddresses.value.length === 0) await saveShippingAddress(billingAddresses.value[0], true);
+
+    await setClientCheckoutAddress();
+    await fetchShippingAndPaymentMethods();
+  } catch (error) {
+    useHandleError(error as ApiError);
+  }
+};
 
-const shippingMethods = computed(() => shippingProviderGetters.getShippingProviders(shippingMethodData.value));
-const paymentMethods = computed(() => paymentMethodData.value);
+onNuxtReady(async () => {
+  if (cartIsEmpty.value) await navigateTo(localePath(paths.cart));
+  isAuthorized.value ? await handleAuthUserInit() : await handleGuestUserInit();
+});
+
+const scrollToTerms = () => {
+  scrollToHTMLObject(ID_CHECKBOX);
+  setShowErrors(true);
+};
+
+const readyToOrder = async () => {
+  if (interactionDisabled.value) return false;
+
+  if (cartIsEmpty.value) {
+    await navigateTo(localePath(paths.cart));
```

Comment:
```suggestion
    send({ type: 'neutral', message: t('emptyCartNotification') });
    await navigateTo(localePath(paths.cart));
```

moved the message from cart to checkout to avoid the message when navigating to the cart directly. So we have to fire the notification here